string_manipulation.rb

```diff
@@ -1,32 +1,215 @@
+# require 'pry'
+
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  puts "NOT IMPLEMENTED"
+  # puts "NOT IMPLEMENTED"
+  str_len = my_string.length
+
+  if my_string.length <= 1
+    return my_string
+  end
+
+  # indexes of vals being swapped
+  idx1 = 0
+  idx2 = str_len - 1
+
+# iterate through first half and last half of string
+  while idx1 < idx2
+    temp = my_string[idx1]
+    my_string[idx1] = my_string[idx2]
+    my_string[idx2] = temp
+
+    idx1 += 1
+    idx2 -= 1
+  end
+
+  return my_string
 end
 
 # A method to reverse each word in a sentence, in place.
 def reverse_words(my_words)
-  puts "NOT IMPLEMENTED"
+  # puts "NOT IMPLEMENTED"
+
+  # store idx of beginning of word
+  # iterate through word string until find whitespace
+  # store idx - 1 of whitespace--this is the end of the word
+  # reverse word in place
+
+  words_len = my_words.length
+
+  if words_len <= 1
+    return my_words
+  end
+  # binding.pry
+
+  word_begin = 0
+  word_end = 0
+
+  # check for string beginning w/whitespace
+  while my_words[word_begin] == " "
+    word_begin += 1
+  end
+
+  (word_begin...words_len).each do |idx|
+    char = my_words[idx]
+
+    # check for whitespace or end of string
+    if char == " " || idx == words_len - 1
+      # this is the end of the word; store idx - 1 or idx if end of string
+        word_end = idx == words_len - 1 ? idx : idx - 1
+
+        # reverse the found word in place
+        while word_begin < word_end
+          temp = my_words[word_begin]
+          my_words[word_begin] = my_words[word_end]
+          my_words[word_end] = temp
+
+          word_begin += 1
+          word_end -= 1
+        end
+
+        # set word_begin to be the current idx + 1
+        word_begin = idx + 1
+    end
+  end
+
+  return my_words
 end
 
 # A method to reverse the words in a sentence, in place.
 def reverse_sentence(my_sentence)
-  puts "NOT IMPLEMENTED"
+  # puts "NOT IMPLEMENTED"
+  full_reverse = string_reverse(my_sentence)

```

Comment:
Since this is all happening in place, you don't need to return anything in any of the methods and this method could simply be:
` string_reverse(my_sentence)`
`reverse_words(my_sentence)`